### ISSUE TITLE
After BroadcastAttestation failed, beacon chain no longer synced

### ISSUE THREAD

**Issue by SimonSMH1015** [prysmaticlabs/prysm#12255]
# 🐞 Bug Report

### Description
Hi

After met the problem below, my beacon stopped synced 
![image](https://user-images.githubusercontent.com/39431803/230821930-8a27d476-5ee1-4321-9ca1-7ea92fe541ba.png)

and kept resyncing.
![image](https://user-images.githubusercontent.com/39431803/230822098-c479532c-891a-4079-a1e8-fcd41e397584.png)
![image](https://user-images.githubusercontent.com/39431803/230822111-531a92d5-0d3e-49eb-a7b8-2f9c95b5e356.png)

### Has this worked before in a previous version?

Find this problem after I updated to v4.0.1.

## 🔬 Minimal Reproduction

I closed '13000' or 30310 port TCP Income as I did  not want any untrusted peers conncted to my node. The beacon command is below: 

beacon-chain --datadir /workdir/consensus --execution-endpoint http://127.0.0.1:8551 --grpc-gateway-host {private-ip} --p2p-local-ip 0.0.0.0 --p2p-host-ip {private-ip} --p2p-tcp-port 30310 --p2p-udp-port 30311 --grpc-gateway-port 30312 --jwt-secret /workdir/execution/jwtsecret --genesis-beacon-api-url https://beaconstate.info --checkpoint-sync-url https://beaconstate.info --bootstrap-node enr:-LG4QLbPQK3_UKE_pGk0VqdkpxVK2tYcu-u3cWKOcEQ2YOvjAxSUGfirTNosY9WsJFyXLItq8hHj98kmqVpJjqyap1uGAYdcLjy7h2F0dG5ldHOIAAAAAAAAAACEZXRoMpD1pf1CAAAAAP__________gmlkgnY0gmlwhAr3Cx-Jc2VjcDI1NmsxoQPlIFzhAeDjeCO6dqJ76c8nEY9yH81rSuAlVJM5AJfmWoN1ZHCCD6I,enr:-KG4QOtcP9X1FbIMOe17QNMKqDxCpm14jcX5tiOE4_TyMrFqbmhPZHK_ZPG2Gxb1GE2xdtodOfx9-cgvNtxnRyHEmC0ghGV0aDKQ9aX9QgAAAAD__________4JpZIJ2NIJpcIQDE8KdiXNlY3AyNTZrMaEDhpehBDbZjM_L9ek699Y7vhUJ-eAdMyQW_Fil522Y0fODdGNwgiMog3VkcIIjKA,enr:-KG4QL-eqFoHy0cI31THvtZjpYUu_Jdw_MO7skQRJxY1g5HTN1A0epPCU6vi0gLGUgrzpU-ygeMSS8ewVxDpKfYmxMMGhGV0aDKQtTA_KgAAAAD__________4JpZIJ2NIJpcIQ2_DUbiXNlY3AyNTZrMaED8GJ2vzUqgL6-KD1xalo1CsmY4X1HaDnyl6Y_WayCo9GDdGNwgiMog3VkcIIjKA,enr:-Ku4QImhMc1z8yCiNJ1TyUxdcfNucje3BGwEHzodEZUan8PherEo4sF7pPHPSIB1NNuSg5fZy7qFsjmUKs2ea1Whi0EBh2F0dG5ldHOIAAAAAAAAAACEZXRoMpD1pf1CAAAAAP__________gmlkgnY0gmlwhBLf22SJc2VjcDI1NmsxoQOVphkDqal4QzPMksc5wnpuC3gvSC8AfbFOnZY_On34wIN1ZHCCIyg,enr:-Ku4QP2xDnEtUXIjzJ_DhlCRN9SN99RYQPJL92TMlSv7U5C1YnYLjwOQHgZIUXw6c-BvRg2Yc2QsZxxoS_pPRVe0yK8Bh2F0dG5ldHOIAAAAAAAAAACEZXRoMpD1pf1CAAAAAP__________gmlkgnY0gmlwhBLf22SJc2VjcDI1NmsxoQMeFF5GrS7UZpAH2Ly84aLK-TyvH-dRo0JM1i8yygH50YN1ZHCCJxA,enr:-Ku4QPp9z1W4tAO8Ber_NQierYaOStqhDqQdOPY3bB3jDgkjcbk6YrEnVYIiCBbTxuar3CzS528d2iE7TdJsrL-dEKoBh2F0dG5ldHOIAAAAAAAAAACEZXRoMpD1pf1CAAAAAP__________gmlkgnY0gmlwhBLf22SJc2VjcDI1NmsxoQMw5fqqkw2hHC4F5HZZDPsNmPdB1Gi8JPQK7pRc9XHh-oN1ZHCCKvg,enr:-Jq4QItoFUuug_n_qbYbU0OY04-np2wT8rUCauOOXNi0H3BWbDj-zbfZb7otA7jZ6flbBpx1LNZK2TDebZ9dEKx84LYBhGV0aDKQtTA_KgEAAAD__________4JpZIJ2NIJpcISsaa0ZiXNlY3AyNTZrMaEDHAD2JKYevx89W0CcFJFiskdcEzkH_Wdv9iW42qLK79ODdWRwgiMo,enr:-Jq4QN_YBsUOqQsty1OGvYv48PMaiEt1AzGD1NkYQHaxZoTyVGqMYXg0K9c0LPNWC9pkXmggApp8nygYLsQwScwAgfgBhGV0aDKQtTA_KgEAAAD__________4JpZIJ2NIJpcISLosQxiXNlY3AyNTZrMaEDBJj7_dLFACaxBfaI8KZTh_SSJUjhyAyfshimvSqo22WDdWRwgiMo,enr:-Ku4QHqVeJ8PPICcWk1vSn_XcSkjOkNiTg6Fmii5j6vUQgvzMc9L1goFnLKgXqBJspJjIsB91LTOleFmyWWrFVATGngBh2F0dG5ldHOIAAAAAAAAAACEZXRoMpC1MD8qAAAAAP__________gmlkgnY0gmlwhAMRHkWJc2VjcDI1NmsxoQKLVXFOhp2uX6jeT0DvvDpPcU8FWMjQdR4wMuORMhpX24N1ZHCCIyg,enr:-Ku4QG-2_Md3sZIAUebGYT6g0SMskIml77l6yR-M_JXc-UdNHCmHQeOiMLbylPejyJsdAPsTHJyjJB2sYGDLe0dn8uYBh2F0dG5ldHOIAAAAAAAAAACEZXRoMpC1MD8qAAAAAP__________gmlkgnY0gmlwhBLY-NyJc2VjcDI1NmsxoQORcM6e19T1T9gi7jxEZjk_sjVLGFscUNqAY9obgZaxbIN1ZHCCIyg,enr:-Ku4QPn5eVhcoF1opaFEvg1b6JNFD2rqVkHQ8HApOKK61OIcIXD127bKWgAtbwI7pnxx6cDyk_nI88TrZKQaGMZj0q0Bh2F0dG5ldHOIAAAAAAAAAACEZXRoMpC1MD8qAAAAAP__________gmlkgnY0gmlwhDayLMaJc2VjcDI1NmsxoQK2sBOLGcUb4AwuYzFuAVCaNHA-dy24UuEKkeFNgCVCsIN1ZHCCIyg,enr:-Ku4QEWzdnVtXc2Q0ZVigfCGggOVB2Vc1ZCPEc6j21NIFLODSJbvNaef1g4PxhPwl_3kax86YPheFUSLXPRs98vvYsoBh2F0dG5ldHOIAAAAAAAAAACEZXRoMpC1MD8qAAAAAP__________gmlkgnY0gmlwhDZBrP2Jc2VjcDI1NmsxoQM6jr8Rb1ktLEsVcKAPa08wCsKUmvoQ8khiOl_SLozf9IN1ZHCCIyg,enr:-LK4QA8FfhaAjlb_BXsXxSfiysR7R52Nhi9JBt4F8SPssu8hdE1BXQQEtVDC3qStCW60LSO7hEsVHv5zm8_6Vnjhcn0Bh2F0dG5ldHOIAAAAAAAAAACEZXRoMpC1MD8qAAAAAP__________gmlkgnY0gmlwhAN4aBKJc2VjcDI1NmsxoQJerDhsJ-KxZ8sHySMOCmTO6sHM3iCFQ6VMvLTe948MyYN0Y3CCI4yDdWRwgiOM,enr:-LK4QKWrXTpV9T78hNG6s8AM6IO4XH9kFT91uZtFg1GcsJ6dKovDOr1jtAAFPnS2lvNltkOGA9k29BUN7lFh_sjuc9QBh2F0dG5ldHOIAAAAAAAAAACEZXRoMpC1MD8qAAAAAP__________gmlkgnY0gmlwhANAdd-Jc2VjcDI1NmsxoQLQa6ai7y9PMN5hpLe5HmiJSlYzMuzP7ZhwRiwHvqNXdoN0Y3CCI4yDdWRwgiOM --p2p-priv-key /workdir/consensus/p2p-priv-key --block-batch-limit 512 --block-batch-limit-burst-factor 3 --p2p-max-peers 1000 --accept-terms-of-use --enable-debug-rpc-endpoints

## 🔥 Error

<pre><code>

I uplaod the full logs at that time , and hopfully that will help,

</code></pre>

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
CentOs
  </code>
</pre>

**What version of Prysm are you running? (Which release)**

<pre>
  <code>
v4.0.1
  </code>
</pre>

**Anything else relevant (validator index / public key)?**

[log-first.txt](https://github.com/prysmaticlabs/prysm/files/11187476/log-first.txt)

**Comment by nisdas** [comment]
@JAVAN000B Try running prysm without these flags:
```
--block-batch-limit 512 --block-batch-limit-burst-factor 3 
```
it appears that you have lost all your peers for some reason

**Comment by JAVAN000B** [comment]
@nisdas Thank you for your reply.  I  will delete those flag and test again. Just want to know why this parameters should not be modified and may cause my node lost all other peers.  I think those 2 flags will affects only when my beacon is in `initial-sync` state. When I met this problem, my node has already fully synced.  And my beacon client was continously having more than 10 outbound peers. So, this really confuesd ,why beacon chain suddenly no longer synced and kept resynced.

**Comment by nisdas** [comment]
@JAVAN000B Do you use any of the api endpoints independently from the beacon node ?

**Comment by nisdas** [comment]
Ok you can run prysm with `--pprof` and the next time you have this issue, can you run this ?
```
http://localhost:6060/debug/pprof/goroutine?debug=1
```
Then paste the output here, it looks like your node is getting stuck for some reason

**Comment by JAVAN000B** [comment]
@nisdas  I noticed that prysm has a new release now, should I updated to that new one? After restart my prysm with debug level, I found my prysm log had some `could not handle p2p pubsub` errors.  I find the new version fixs some Block construction and Block processing tings, are these related?

**Comment by nisdas** [comment]
Yes, feel free to update @JAVAN000B , lets us know if it fixes your issues

**Comment by peterfo-tr** [comment]
FYI, we're still seeing this error after upgrading to prysm-beacon-chain to v4.0.2

**Comment by nisdas** [comment]
@peterfo-tr Do you have more logs, or the profiles that are mentioned here ?
https://github.com/prysmaticlabs/prysm/issues/12255#issuecomment-1505088166

**Comment by peterfo-tr** [comment]
> @peterfo-tr Do you have more logs, or the profiles that are mentioned here ? [#12255 (comment)](https://github.com/prysmaticlabs/prysm/issues/12255#issuecomment-1505088166)

@nisdas we observed the behaviour with both v4.0.1 and v4.0.2 on both göerli-prater and mainnet. Currently we are running with `--pprof` on our göerli-prater deployments.

**Comment by nisdas** [comment]
@peterfo-tr Thanks for the response, in the event that you do come across this again can you fetch a profile ?
```
curl http://localhost:6060/debug/pprof/goroutine?debug=1 -o goroutine.txt
```
This would help in diagnosing the issue on our end

**Comment by nisdas** [comment]
Thanks @peterje-tr , do you also have the logs at this time ?

**Comment by nisdas** [comment]
hey @peterje-tr , I have tried investigating it but haven't had much luck. The logs would indicate that there was an abrupt drop in network ingress for discovery, Is it possible its a networking issue on your host machine ? If you think that isn't the case, the last option would be to enable `--verbosity=trace`. However this will pretty much blow up your logs, as it will log out everything. Once you have a failure again, you can paste those logs here again.

**Comment by JAVAN000B** [comment]
Thanks, @nisdas. After I updated to v4.0.2, this problem occures again. Unfortunatly, I could not get the goroutine.txt on my machine because of the cloud server provider nor open trace log : (  But this problem appears again on v4.0.2. I also notice that this problem always happens after beacon run smoothly 2 or 3 days , hopefully, this information will help.

**Comment by JAVAN000B** [comment]
hello @nisdas . I think I meet this problem again on my test environment . One bad thing is I forgot to open pporf on my beacon client. But I do have the trace level log and it is posted below. 
### The problem shows between 16:02:13 and 17:15:24 ###
My beacon client suddenly kept printed this error and no longer synced new payload.
![image](https://user-images.githubusercontent.com/39431803/233602954-f4fec718-454e-442c-963b-6367bcf3220f.png)
And then, the BroadcastAttestation error appeared.
![image](https://user-images.githubusercontent.com/39431803/233603580-97e91851-19e4-4b38-b1ec-dcbe7d1b4665.png)
Last, it started resyncing.
![image](https://user-images.githubusercontent.com/39431803/233603852-33c1f250-e103-475a-b83c-f5765ecb2948.png)

BTw, just mentioned, I am doing this experiment in order to find why sometimes the rpc request got  `i/o deadline exeed` error. This makes my peers were considerd as bad peers when the nodes my peers connected to doing revalidpeer function, as the bad respond increased.
Is there any way I can aviod this `i/o deadline exeed` error or avoid my node been banned from others because of 'bad' response?

The  log file is too large, could you provide a way I can upload?

**Comment by nisdas** [comment]
@JAVAN000B How big is the file ? If you compress it via tar, what size do you get

**Comment by simonSun1015** [comment]
[question1.zip](https://github.com/prysmaticlabs/prysm/files/11302956/question1.zip)
[question2.zip](https://github.com/prysmaticlabs/prysm/files/11302957/question2.zip)
[question3.zip](https://github.com/prysmaticlabs/prysm/files/11302958/question3.zip)
@nisdas Im JAVAN0000B, Finally, I divided the log into 3 parts and upload

**Comment by nisdas** [comment]
@JAVAN000B This appears to be an issue with your geth node:
```
time="2023-04-21 16:02:32" level=error msg="Could not handle p2p pubsub" error="could not process block: could not validate new payload: timeout from http.Client: received an undefined execution engine error" prefix=sync topic="/eth2/bba4da96/beacon_block/ssz_snappy"
```
This error only comes about when your execution client can't process the payload in time. It might be best to look there to see why it doesnt work

**Comment by nisdas** [comment]
Yeah, a HDD for geth will cause issues. I would look there to see if that are any logs that might explain why geth is processing payloads so slowly.

**Comment by MrDonalds** [comment]
i met the same problem

the inbound  connection drop and then after about 20 min inbound went to 0，and  beacon failed to sync

![image](https://user-images.githubusercontent.com/12430046/234203076-0984192c-1a2a-4283-b01f-01718ab7a785.png)


![4943A7D9-1033-4E97-BE0A-2C3326B7A5F9](https://user-images.githubusercontent.com/12430046/234201728-d312cfc1-3d81-4c95-b02d-3fcc6b9a7a67.png)

**Comment by nisdas** [comment]
@MrDonalds What flags were you running with ? For the beacon node

**Comment by JAVAN000B** [comment]
@nisdas I think my beacon no longer synced is beacause every message from gossip net were ignored.

**Comment by JAVAN000B** [comment]
My log are full of this things.
![image](https://user-images.githubusercontent.com/39431803/234783356-b80fd77e-7940-4519-b7f5-bf9e63e38040.png)

@nisdas hello. I watched the log again, and I think my beacon was no longer synced is because it ignored all meassage from gossip network. Do you have any ideas about why this things happens and why for some reason, it stuck and ignore all messages?

**Comment by nisdas** [comment]
@JAVAN000B Trace logs are very noisy, so its hard to know what is the root cause here. Those logs don't say anything much, like I said earlier there possibly is an issue with your execution client from your earlier logs

**Comment by nisdas** [comment]
@peterje-tr @JAVAN000B 

Are you guys still running into this issue ?

**Comment by JAVAN000B** [comment]
@nisdas yes, we still facing this problem. I used SSD but this one again happended after running 7 days. Unfortunately, this time I did not open `--pprof` neither  `trace`

**Comment by regularcoffeecup** [comment]
@nisdas @JAVAN000B I am having this problem. I am using and AVADO with SSD. Unlike others my inbound count doesn't seem to drop, but it does seem to compact the database too often. I can provide more info. I was running for a year without issues from april to april.

**Comment by JAVAN000B** [comment]
@regularcoffeecup hello! you mentioned that it does seem to compact the database too often. I want to know how could you get this. Is there any target words in logs? or there are some metrics? I could check my logs  and nodes to see if I got the same errors.

**Comment by prestonvanloon** [comment]
Maybe unrelated, but I was discussing with @james-prysm another issue with an Avado user that seemed to point to the system running low on available RAM and starting to use the swap space on disk. Using swap could lead to slow CPU operations and potentially causes issues like the observed issues above. 

Is there a way that someone could report the current memory usage of the system at the time they are experiencing these issues?

**Comment by prestonvanloon** [comment]
Also have 1000 peers could be part of the problem... Have you tried lowering that to a reasonable maximum?

**Comment by regularcoffeecup** [comment]
@prestonvanloon  and @JAVAN000B  Thanks for responding. Really appreciate it. Please see above for current comment I forgot to tag you.

**Comment by JAVAN000B** [comment]
@prestonvanloon I am running my node with 32gb RAM and SSD, in total geth and prysm only used half of that, for prysm only less than 4gb was used.  As to the number 1000, yes, I know this is some how "hilarious". The reason why I set this is I want my other nodes are able to connect to this one at any time rather than received a `too many peers` goodbye message.   This used to work on version 3.0s but after I updated to 4.0.0 I met this problem.  But anyway ,I will try set p2p-max-peers to its default value 45 and see if this happens again.

**Comment by JAVAN000B** [comment]
@prestonvanloon thank you for suggestions. And could you mind I take up a little of your time?   Yes, --peer solves many problems. From my understanding, If I want to have a priority connection between A and B node, then I should add --peer each node on both sides. But there is a situation that after 3 nodes are peered by setting --peer. I want my fourth peer to join them. Now, the only way to achieve this is to add one more `--peer` flag in commands for each of nodes and stop then restart previous three nodes. For some reason, e.g. stacking I cannot restart previous nodes, so I use that 1000 max peers method. 

Not sure if there are some dynamic adding peers methods in Prysm, just like admin.addPeers() in geth.  This discussion may be a bit off topic 😁

**Comment by prestonvanloon** [comment]
@JAVAN000B we don't have an API like geth does for admin_addPeer or admin_addTrustedPeer at this time, but I understand your use case. I'll discuss with the team and we can file a feature request. Thanks

**Comment by nisdas** [comment]
@regularcoffeecup It seems like your issue is different to the other cases, as in these cases it goes down to 0 peers while you still have peers on your node. Your issue might be more of geth hogging the CPU during database compactions which cause issues with Prysm being able to perform its duties on time.

**Comment by LilkkJerry** [comment]
> @JAVAN000B we don't have an API like geth does for admin_addPeer or admin_addTrustedPeer at this time, but I understand your use case. I'll discuss with the team and we can file a feature request. Thanks

I've been watching this issue for weeks, thanks for proposing this, I believe it would be very helpful

**Comment by MrDonalds** [comment]
> Maybe unrelated, but I was discussing with @james-prysm another issue with an Avado user that seemed to point to the system running low on available RAM and starting to use the swap space on disk. Using swap could lead to slow CPU operations and potentially causes issues like the observed issues above.
> 
> Is there a way that someone could report the current memory usage of the system at the time they are experiencing these issues?

my node is not 32GB, i am runing a machine with 16GB RAM with 4GB SWAP, the memory usage of the machine is nearly 90%, would this cause the problem?

**Comment by regularcoffeecup** [comment]
@MrDonalds My node has 32GB and often runs at 40 percent so I think 90% could be expected at times on 16gb but not all the time... deleting all my packages and re downloading them helped but it did take me a week to sync. Also putting a fan on the avado helped, but depending on your problems these could help with nothing.

**Comment by nisdas** [comment]
I think this has been resolved since v4.0.6 fully with @SimonSMH1015 help. Closing this issue unless someone feels that is not the case.
